sources/web/datalab/polymer/components/auth-panel/auth-panel.ts

```diff
@@ -64,8 +67,11 @@ class AuthPanel extends Polymer.Element {
   _signInChanged(signedIn: boolean) {
     this._signedIn = signedIn;
     if (signedIn) {
-      this._userInfo = 'Signed in as ' + GapiManager.getSignedInEmail();
-      this._projectInfo = 'No project is set';  // TODO
+      GapiManager.getSignedInEmail()
+        .then((email: string) => {
+          this._userInfo = 'Signed in as ' + email;
+          this._projectInfo = 'No project is set';  // TODO
+        });
     }
     const ev = new Event('signInOutDone');
     this.dispatchEvent(ev);
```

Comment:
I think your analysis is correct, but it's good to put code in the right place. Done.